inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -545,7 +543,24 @@ public void writeMessage(InputStream message) {
           clientStream.statsTraceCtx.inboundMessage(outboundSeqNo);
           clientStream.statsTraceCtx.inboundMessageRead(outboundSeqNo, -1, -1);
 
-          long messageLength = getMessageLength(message);
+          /*long messageLength = serverAssumedMessageSize == -1
+                  ? getMessageLength(message) : serverAssumedMessageSize;*/
+          long messageLength = -1;
+          try {
+            if (!(message instanceof KnownLength) && !(message instanceof ByteArrayInputStream)) {
```

Comment:
It should be inside this if that you check for assumedMessageSize.  If it is -1 do the existing logic otherwise just set messageLength to assumedMessageSize.

Same for the client stream.